libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -1137,7 +1141,6 @@ uint16_t x, usDataLength, usQuestions;
 
 					pucByte += sizeof( DNSAnswerRecord_t ) + sizeof( uint32_t );
 					uxSourceBytesRemaining -= ( sizeof( DNSAnswerRecord_t ) + sizeof( uint32_t ) );
-					break;
```

Comment:
This is the break after seeing the first response.
In other places, you're testing if the new feature of multiple addresses is used.
Can you do the same with this break;
~~~c
#if( ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY <= 1 )
    /* One response was received, which is enough for this configuration. */
    break;
#endif
~~~

Reply:
My initial version of the code did this!  I felt it might be confusing so I opted instead to let the new test at line 1082 exit the loop in this case.  I do see your point for readers evaluating the change who aren't using the feature.  If you think it will help existing users, I will restore the `#if` clause as suggested.

Reply:
Have made the change as suggested.